scripts/tests/test_migrate_metadata.py

```diff
@@ -0,0 +1,145 @@
+from nose.tools import *  # noqa
+from tests.base import OsfTestCase
+
+import json
+from modularodm import Q
+
+from framework.mongo.utils import to_mongo
+
+from website.project.model import ensure_schemas, MetaSchema
+from tests import factories
+from scripts.migration.migrate_registered_meta import main as do_migration
+from scripts.migration.migrate_registered_meta import get_old_registered_nodes
+
+SCHEMA_NAMES = [
+    'Open-Ended Registration',
+    'OSF-Standard Pre-Data Collection Registration',
+    'Replication Recipe (Brandt et al., 2013): Pre-Registration',
+    'Replication Recipe (Brandt et al., 2013): Post-Completion'
+]
+OLD_META = {
+    'Open-Ended Registration': {
+        'summary': 'some airy',
+    },
+    'OSF-Standard Pre-Data Collection Registration': {
+        'comments': 'Standard',
+        'datacompletion': 'Yes',
+        'looked': 'Yes',
+    },
+    'Replication Recipe (Brandt et al., 2013): Pre-Registration': {
+        'item1': 'Ver',
+        'item10': 'yes',
+        'item11': 'fas',
+        'item12': 'afs',
+        'item13': 'fsa',
+        'item14': 'fsa',
+        'item15': 'fsa',
+        'item16': 'sf',
+        'item17': 'Exact',
+        'item18': 'Different',
+        'item19': 'Different',
+        'item2': 'vsf',
+        'item20': 'Different',
+        'item21': 'Close',
+        'item22': 'Exact',
+        'item23': 'Exact',
+        'item24': 'fsasf',
+        'item25': 'asfsfa',
+        'item26': 'safasf',
+        'item27': 'asf',
+        'item28': 'fassf',
+        'item3': 'fafa',
+        'item4': 'fsafds',
+        'item5': 'fafa',
+        'item6': 'asdfsadf',
+        'item7': 'sfsaf',
+        'item8': 'sfdsdf',
+        'item9': 'sfd',
+    },
+    'Replication Recipe (Brandt et al., 2013): Post-Completion': {
+        'item29': 'adad',
+        'item30': 'asd',
+        'item31': 'asd',
+        'item32': 'not significantly different from the original effect size',
+        'item33': 'informative failure to replicate',
+        'item34': 'asdasd',
+        'item35': 'ds',
+        'item36': 'ads',
+        'item37': 'das',
+    },
+}
+
+
+class TestMigrateSchemas(OsfTestCase):
+    def setUp(self):
+        super(TestMigrateSchemas, self).setUp()
+
+        MetaSchema.remove()
+        ensure_schemas()
+
+        self.open_ended_schema = MetaSchema.find_one(
```

Comment:
Also, you can DRY this setup method a bit by using a helper method on the test class. (as long as it doesn't start with the word "test", it won't be run as a unit test) A lot of this code is repeated 4x.

For example,

```

self.open_ended = self._reg_from_schema_name(SCHEMA_NAMES[0])

def _reg_from_schema_name(self, name):
    schema = MetaSchema.find_one(
            Q('name', 'eq', name) &
            Q('schema_version', 'eq', 1)
        )
        registration_obj = factories.RegistrationFactory(
            registered_meta={
                to_mongo(name): json.dumps(OLD_META[name])
            },
            registered_schema=schema
        )
        del registration_obj.registered_meta['Template1']
        registration_obj.save()
    return registration_obj
```